### ISSUE TITLE
Attuned items get filtered with unattuned items sometimes.

### ISSUE THREAD

**Issue by cfedey** [imevul/ArkInventory_Attune#3]
Ever since the last update, occasionally when attuning an item it will stay in the category I have unattuned items assigned to instead of moving to the category for attuned items. These will be items that I've attuned for the first time that don't have affixes; just normal raid epics. Attached image shows an example of this. The issue persists after refreshing bags until I either relog or /reload.
![image](https://github.com/user-attachments/assets/ffb4a7e7-2397-4a9a-a61a-6e7b9b0bc1cf)

These are my current rules:
![image](https://github.com/user-attachments/assets/3b314c5b-d5f5-416e-9fb5-d856988ecff1)

"Attunable" is `(not attuned() and attunable()) or (not attuned() and partiallyattuned())`.
"Attuned" is just `attuned()`.

Issue occurs even when ArkInventory is the only active addon. No LUA errors are thrown when it happens.

**Comment by imevul** [comment]
Are the items *forged?

Also, I would have done the priority differently. Lower priority number = rule has higher priority / gets evaluated first, so if you set attuned() to have higher priority (as in lower priority number), you can simplify attunable to be `attunable()`. I'm assuming "AccAccunable" is `partiallyattuned()` here.

**Comment by cfedey** [comment]
None of those two items in the image were forged, no. I can't say if I have or haven't seen it with forged items, but if I do I'll update. 

I used to have the Attuned rule at the bottom, but I was messing with things to see if it had any affect on the issue. AccAccunable (which should be AccAttunable, oops) is disabled so it shouldn't matter, but it is `attunableatall() and not attunablebyme() and not attuned()`. I made that for someone else who was trying to sort other-character-attunable stuff a while back.

**Comment by cfedey** [comment]
More info:
I tried using ArkInventory's "instant sort" option, and the issue still occurs.

![image](https://github.com/user-attachments/assets/35348049-bacf-476b-b572-029418d60e6e)

Before /reload:
![image](https://github.com/user-attachments/assets/8f07e663-a851-45bf-8dfe-6a80789a09b4)

After /reload:
![image](https://github.com/user-attachments/assets/0320989a-6834-42c0-9d65-fca39b9befc9)

**Comment by imevul** [comment]
So, if I got this correct. What happens is:
1. Your rules are, in priority order:
    a. Attuned: `attuned()`
    b. Attunable: `(not attuned() and attunable()) or (not attuned() and partiallyattuned())`
2. You have an unattuned item in your bags, sorted into Attunable rule
3. You attune the item and put it back in your bags
4. It's still sorted into Attunable rule
5. You /reload and the item is now correctly sorted into Attuned rule.

**Comment by cfedey** [comment]
> So, if I got this correct. What happens is:
> 
>     1. Your rules are, in priority order:
>        a. Attuned: `attuned()`
>        b. Attunable: `(not attuned() and attunable()) or (not attuned() and partiallyattuned())`
> 
>     2. You have an unattuned item in your bags, sorted into Attunable rule
> 
>     3. You attune the item and put it back in your bags
> 
>     4. It's still sorted into Attunable rule
> 
>     5. You /reload and the item is now correctly sorted into Attuned rule.

Yes, though I changed the Attunable rule to `not attuned() and (attunable() or partiallyattuned())`. Dunno if the addon likes that better, but it looks cleaner at least.
